### ISSUE TITLE
The Performance of  the Deployed Model on Android is Far from What on the PC

### ISSUE THREAD

**Issue by joeshow79** [pytorch/android-demo-app#195]
Hi, 
    I trained one model to detect the steel rebar base on yolov5x model. The testing result is good on PC. And I followed the guide (https://github.com/pytorch/android-demo-app/pull/185) to convert  the model to torchscript model (ptl) and integrate it to the demo app. Then the demo app could work and output the result, but there is huge gap between the results on PC and app, see below pic for comparison. 

Result on PC (confidence thresh is 0.25)
![image](https://user-images.githubusercontent.com/15626897/138205624-c55fc0c9-65d9-47fe-b9ad-2e41fb467f5b.png)
Result on App(confidence thresh is 0.2)
![image](https://user-images.githubusercontent.com/15626897/138205741-346258cf-ce09-4d41-b62f-eb6afba8d0b0.png)
I also tuned the --optimze when export the model
python3 /workspace/src/github/yolov5/export.py --weight runs/train/exp32/weights/best.pt --include torchscript --optimize
But there is no significant difference after the tuning. So far have no more clue to figure out ...

Any tips or suggestion is appreciated, thanks!

**Comment by joeshow79** [comment]
Found there are some relation with the code
1. nNmsLimit in PrePostProcess.java default is 15, which is not good for mass object detection.
2. Confidence filter and NMS filter share the same threshold (mThreshold in PrePostProcess.java) which may not good to tune the performance.

After fix the above 2 issues, the output got better, but still not as good as the result from the inference on PC, 3 significant issues remain
1. The confidence of the proposal BBox is very small (distribute from 0.) compared to the inference result on PC
2. Many False Positive with hight confidence (>0.9) and small rect (suspect the rect size is negative)
3. Still many False Negative(recall is not good)
![image](https://user-images.githubusercontent.com/15626897/141954423-6d6d051d-def1-4974-9ac8-ff57eb28de4a.png)

**Comment by joeshow79** [comment]
Finally work out the reason, 
1. The dimension of the output predicted tensor is wrong, the class is 1, so the dimension for 1 predict is 6 instead of 5(wrong)
2. Increase the maximum detection number to 1000(from 15)
And one minor change to make the final conf by conf = object_conf * cls_conf

**Comment by Karl8** [comment]
Thank you joeshow79. I had the same issue, and I solved it by:
1. Increace the mNmsLimit to 300.
2.Padding the input image with rgb=(114, 114, 114) before resizing it, like the yolov5 PC version in dataset.py:222. The function letterbox() will resize and pad image while meeting stride-multiple constraints
